walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,80 @@
 
 package com.google.sps;
 
+import java.lang.Math;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Collect parameters of meeting request
+    Collection<String> attendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getAttendees();
+    int duration = (int) request.getDuration();
+
+    List<TimeRange> viableTimes = new ArrayList<>();
+    List<TimeRange> reservedTimes = new ArrayList<>();
+
+    // List of pre-existing events sorted by starting time
+    events.stream()
+          .sorted((e1, e2) -> e1.getWhen().start() - e2.getWhen().start())
+          .filter (e -> !Collections.disjoint(e.getAttendees(), attendees))
+          .forEach(e -> {
```

Comment:
nit: Instead of defining the list and then adding to it here, you can use a collector to build the list directly. https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toList--